torchtext/csrc/gpt2_bpe_tokenizer.cpp

```diff
@@ -282,14 +364,47 @@ std::vector<int64_t> GPT2BPEEncoder::Encode(const std::string& text) {
 std::vector<std::string> GPT2BPEEncoder::Tokenize(const std::string& text) {
   std::vector<std::string> bpe_tokens;
   for (const auto& token : PreTokenize_(text)) {
-    auto byte_encoded_token = ByteEncode_(token);
+    bool is_never_split_token =
+        bpe_never_split_set_.find(token) != bpe_never_split_set_.end();
+    auto byte_encoded_token = ByteEncode_(token, is_never_split_token);
     for (const auto& bpe_token : BPE_(byte_encoded_token)) {
       bpe_tokens.push_back(bpe_token);
     }
   }
   return bpe_tokens;
 }
 
+int64_t GPT2BPEEncoder::AddSpecialTokens(
+    const c10::Dict<std::string, std::string>& standard_special_tokens_dict,
+    const std::vector<std::string> additional_special_tokens) {
+  int64_t newly_added = 0;
+
+  for (auto const& token : standard_special_tokens_dict) {
+    if (added_tokens_encoder.contains(token.value()))
+      continue;
+    bpe_never_split_set_.insert(token.value());
+    if (!bpe_encoder_.contains(token.value())) {
+      added_tokens_encoder.insert(
+          token.value(), bpe_encoder_.size() + added_tokens_encoder.size());
+      newly_added++;
+    }
+  }
+
+  for (auto const& token : additional_special_tokens) {
+    if (added_tokens_encoder.contains(token))
+      continue;
+    bpe_never_split_set_.insert(token);
+    if (!bpe_encoder_.contains(token)) {
+      added_tokens_encoder.insert(
+          token, bpe_encoder_.size() + added_tokens_encoder.size());
+      newly_added++;
+    }
+  }
+
+  added_to_vocab_tokens_count += newly_added;
```

Comment:
I might be missing something here but where exactly is this variable used?

Reply:
I originally meant to use this variable to return the vocab size (original encoder size + added encoder size), similar to how HuggingFace API provisions it. But it appears that we don't really have any existing requirement for fetching size, so I just removed it in the latest commit.